pkg/virt-handler/vm.go

```diff
@@ -1207,17 +1208,14 @@ func isGuestAgentSupported(vmi *v1.VirtualMachineInstance, commands []v1.GuestAg
 	}
 
 	if checkSSH && !_guestAgentCommandSubsetSupported(SSHRelatedGuestAgentCommands, commands) {
-		log.Log.V(3).Object(vmi).Info("This guest agent doesn't support required public key commands")
-		return false
+		return false, "This guest agent doesn't support required public key commands"
 	}
 
 	if checkPasswd && !_guestAgentCommandSubsetSupported(PasswordRelatedGuestAgentCommands, commands) {
-		log.Log.V(3).Object(vmi).Info("This guest agent doesn't support required password commands")
-		return false
+		return false, "This guest agent doesn't support required password commands"
 	}
 
-	log.Log.V(3).Object(vmi).Info("This guest agent is supported")
-	return true
+	return true, ""
```

Comment:
Having moved the Log outside of this function you have to return `This guest agent is supported`, otherwise you will lose a log. I think that here you have 2 ways:

- leave this return as is, but you have to restore the log inside this function
- return `This guest agent is supported` but you have to adjust the tests properly

Reply:
Fixed, thanks. I chose to always return some reason even if it is `This guest agent is supported` and adjusted the tests.